frontend/src/features/workspace/components/AdminFeedbackModal/AdminFeedbackModalContent.tsx

```diff
@@ -0,0 +1,175 @@
+import { useCallback, useState } from 'react'
+import { useForm } from 'react-hook-form'
+import { GoThumbsdown, GoThumbsup } from 'react-icons/go'
+import {
+  Flex,
+  Link,
+  ModalBody,
+  ModalCloseButton,
+  ModalHeader,
+  Stack,
+  Text,
+} from '@chakra-ui/react'
+
+import { AdminFeedbackRatingDown, AdminFeedbackRatingUp } from '~shared/types'
+
+import { useIsMobile } from '~hooks/useIsMobile'
+import Button from '~components/Button'
+import IconButton from '~components/IconButton'
+import Textarea from '~components/Textarea'
+
+import { useAdminFeedbackMutation } from '~features/workspace/mutations'
+
+enum FeedbackModalContentState {
+  Rating,
+  CallForComment,
+  CommentBox,
+}
+
+type AdminFeedbackCommentForm = {
+  comment: string
+}
+
+export const AdminFeedbackModalContent = ({
+  onClose,
+}: {
+  onClose: () => void
+}) => {
+  const [contentState, setContentState] = useState(
+    FeedbackModalContentState.Rating,
+  )
+  const [feedbackId, setFeedbackId] = useState('')
+  const { createAdminFeedbackMutation, updateAdminFeedbackMutation } =
+    useAdminFeedbackMutation()
+
+  const handleRatingClick = useCallback(
+    (rating: number) => {
```

Comment:
can be more specific in typing

```suggestion
    (rating: AdminFeedbackRatingUp | AdminFeedbackRatingDown) => {
```

Reply:
Same as below! [632f87e](https://github.com/opengovsg/FormSG/pull/6465/commits/632f87e807e3a074d773673b2453226cbf620df1)